core/src/test/java/org/apache/iceberg/rest/responses/TestConfigResponseParser.java

```diff
@@ -214,4 +214,47 @@ public void roundTripSerdeWithEndpoints() {
     assertThat(ConfigResponseParser.toJson(ConfigResponseParser.fromJson(json), true))
         .isEqualTo(expectedJson);
   }
+
+  @Test
+  public void idempotencyLifetimeOnly() {
+    ConfigResponse response = ConfigResponse.builder().withIdempotencyKeyLifetime("PT30M").build();
+
+    String expectedJson =
+        "{\n"
+            + "  \"defaults\" : { },\n"
+            + "  \"overrides\" : { },\n"
+            + "  \"idempotency-key-lifetime\" : \"PT30M\"\n"
+            + "}";
+
+    String json = ConfigResponseParser.toJson(response, true);
+    assertThat(json).isEqualTo(expectedJson);
+    ConfigResponse roundTripped = ConfigResponseParser.fromJson(json);
+    assertThat(roundTripped.idempotencyKeyLifetime()).isEqualTo("PT30M");
+  }
+
+  @Test
+  public void invalidIdempotencyLifetime() {
+    ConfigResponse parsed =
+        ConfigResponseParser.fromJson(
+            "{\n"
+                + "  \"defaults\" : { },\n"
+                + "  \"overrides\" : { },\n"
+                + "  \"idempotency-key-lifetime\" : \"not-a-duration\"\n"
+                + "}");
+    // invalid value is treated as "not advertised"
```

Comment:
I believe an invalid value should always throw, similar to how e.g. an invalid value for a number throws